libs/ledgerjs/packages/hw-app-eth/src/modules/EIP712/index.ts

```diff
@@ -218,6 +241,109 @@ const EIP712SendStructImplem = async (
   return Promise.resolve();
 };
 
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "activate"
+): Promise<Buffer>;
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "contractName",
+  data: FilteringInfoContractName
+): Promise<Buffer>;
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "showField",
+  data: FilteringInfoShowField
+): Promise<Buffer>;
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "activate" | "contractName" | "showField",
+  data?: FilteringInfoContractName | FilteringInfoShowField
+): Promise<Buffer | void> {
+  enum APDU_FIELDS {
+    CLA = 0xe0,
+    INS = 0x1e,
+    P1 = 0x00,
+    P2_activate = 0x00,
+    P2_contract_name = 0x0f, // officially named "message info"
+    P2_show_field = 0xff,
+  }
+
+  switch (type) {
+    case "activate":
+      return transport.send(
+        APDU_FIELDS.CLA,
+        APDU_FIELDS.INS,
+        APDU_FIELDS.P1,
+        APDU_FIELDS.P2_activate
+      );
+
+    case "contractName": {
+      const { displayName, filtersCount, sig } =
+        data as FilteringInfoContractName;
+      const displayNameLengthBuffer = Buffer.from(
+        intAsHexBytes(displayName.length, 1),
+        "hex"
+      );
+      const displayNameBuffer = Buffer.from(displayName);
+      const filtersCountBuffer = Buffer.from(
+        intAsHexBytes(filtersCount, 1),
+        "hex"
+      );
+      const sigLengthBuffer = Buffer.from(
+        intAsHexBytes(sig.length / 2, 1),
+        "hex"
+      );
+      const sigBuffer = Buffer.from(sig, "hex");
+
+      const callData = Buffer.concat([
+        displayNameLengthBuffer,
+        displayNameBuffer,
+        filtersCountBuffer,
+        sigLengthBuffer,
+        sigBuffer,
+      ]);
+
+      return transport.send(
+        APDU_FIELDS.CLA,
+        APDU_FIELDS.INS,
+        APDU_FIELDS.P1,
+        APDU_FIELDS.P2_contract_name,
+        callData
+      );
+    }
+
+    case "showField": {
+      const { displayName, sig } = data as FilteringInfoShowField;
+      const displayNameLengthBuffer = Buffer.from(
+        intAsHexBytes(displayName.length, 1),
+        "hex"
+      );
+      const displayNameBuffer = Buffer.from(displayName);
+      const sigLengthBuffer = Buffer.from(
+        intAsHexBytes(sig.length / 2, 1),
```

Comment:
Not sure to understand why we divide by 2

Reply:
The signature is an hex string, meaning two caracters = one 4 bits decimal (0-15), so a base 16 number. And the length expected is in base 16.